[#139390] Allow extending reservations inside lock window

jhanggi opened the pull request:
You are now allowed to edit a reservation inside the lock window, but
you cannot change the start time and you cannot shorten the duration,
though you may extend it.

giladshanan commented on spec/app_support/reservations/duration_change_validations_spec.rb:
```diff
@@ -63,16 +61,32 @@
+              reservation.reserve_end_at -= 5.minutes
+              expect(validator).not_to be_valid
+              expect(validator.errors.full_messages)
+                .to include("Reserve end at cannot be shortened inside the lock window")
```
I'm wondering about the wording of the error messages here.  Would 'reservation end time cannot be shortened...' be clearer than 'reserve end at cannot be shortened...' ?

meara replied:
@jhanggi I'm not sure what the history is behind the human-readable version of `reserve_end_at` is. Idk if it's worth changing though (i.e. would it cause more user confusion to change now).

meara commented on app/models/reservation.rb:
```diff
@@ -230,7 +230,7 @@ def reserve_start_at_editable?
 
   def reserve_end_at_editable?
-    outside_lock_window? && Time.zone.now <= reserve_end_at && extendable? && actual_end_at.blank?
+    Time.zone.now <= reserve_end_at && extendable? && actual_end_at.blank?
```
Should this also be `Time.current` since we're using that below?

jhanggi commented:
Good call on the messages. I moved the message to `duration_mins`, and now the message shows "Duration cannot be shortened...". It also highlights the duration field in red. I added the `(X hours)` to make that error message wrap below; otherwise it just overflowed the gray box.

meara commented:
👍  thanks Jason! I think moving the message to duration makes a lot of sense.
